src/qforte/abc/ansatz.py

```diff
@@ -74,21 +74,25 @@ def get_res_over_mpdenom(self, residuals):
             residuals.
         """
 
+        reference_state = qf.QubitBasis(self._nqb)
+        for k, occ in enumerate(self._ref):
+            reference_state.set_bit(k, occ)
+
         resids_over_denoms = []
 
-        # loop over toperators
-        for mu, m in enumerate(self._tops):
-            sq_op = self._pool_obj[m][1]
+        for mu, (index, _) in enumerate(self._excited_dets):
+            excited = qf.QubitBasis(index)
 
-            temp_idx = sq_op.terms()[0][2][-1]
-            if temp_idx < int(sum(self._ref)/2): # if temp_idx is an occupied idx
-                sq_creators = sq_op.terms()[0][1]
-                sq_annihilators = sq_op.terms()[0][2]
-            else:
-                sq_creators = sq_op.terms()[0][2]
-                sq_annihilators = sq_op.terms()[0][1]
+            denom = 0
 
-            denom = sum(self._orb_e[x] for x in sq_annihilators) - sum(self._orb_e[x] for x in sq_creators)
+            for qubit in range(self._nqb):
+                if reference_state.get_bit(qubit):
+                    if not excited.get_bit(qubit):
+                        # Annnihilated
+                        denom += self._orb_e[qubit]
+                elif excited.get_bit(qubit):
+                    # Created
+                    denom -= self._orb_e[qubit]
 
```

Comment:
I have a question here: do we assume a JW transformation here? In the old version based on the sq_operators, the denominator are independent of the fermionic mapping. But if we examine qubits it should depend if we use parity or JW. So if this is the case, I would consider adding a function that computes the denominator and that keeps into account the transformation used. We might have to think about this carefully to make it work in the general case.

Reply:
Strictly speaking, we don't even assume that our Hamiltonian is from a fermionic system.

I think you mean, "if we have a fermionic system, do we need to assume a JW transformation in order for these quantities to truly be MP denominators"? And yes, that's correct.

The code that introduces the assumption is the [`build_orb_energies` function](https://github.com/evangelistalab/qforte/blob/886de7e3c092b2f11decedf31fedff73ebfb9082/src/qforte/abc/ansatz.py#L54-L56), which is only correct in some JW transformation. That was true before this PR.

Let's assume `build_orb_energies` did not introduce this assumption. Does this code deepen it? Unfortunately, yes. It explicitly assumed that every bit corresponds to some orbital.

I thought about the question of how to make something truly general, and I didn't find a satisfying answer.

Reply:
@fevangelista Am I clear to merge this PR in, even with the implicit JW assumption?